backend/src/apis/snapshot_data.rs

```diff
@@ -0,0 +1,249 @@
+use num_bigint::BigInt;
+use std::collections::HashMap;
+
+use halo2_proofs::{
+    halo2curves::bn256::Fr as Fp,
+    plonk::{keygen_pk, keygen_vk},
+};
+use snark_verifier_sdk::CircuitExt;
+
+use summa_solvency::{
+    circuits::{
+        merkle_sum_tree::MstInclusionCircuit,
+        solvency::SolvencyCircuit,
+        utils::{full_prover, generate_setup_params},
+    },
+    merkle_sum_tree::utils::big_int_to_fp,
+    merkle_sum_tree::{Entry, MerkleSumTree},
+};
+
+use crate::apis::csv_parser::parse_csv_to_assets;
+
+struct SnapshotData<
+    const LEVELS: usize,
+    const L: usize,
+    const N_ASSETS: usize,
+    const N_BYTES: usize,
+    const K: u32,
+> {
+    exchange_id: String,
+    mst: MerkleSumTree<N_ASSETS>,
+    assets: Vec<Asset>,
+    proofs_of_inclusion: HashMap<u64, InclusionProof>,
+    proof_of_solvency: Option<SolvencyProof<N_ASSETS>>,
+}
+
+#[derive(Debug, Clone)]
+pub struct Asset {
+    pub name: String,
+    pub pubkeys: Vec<String>,
+    pub balances: Vec<BigInt>,
+    pub sum_balances: Fp,
+    pub signature: Vec<String>,
+}
+
+#[derive(Debug, Clone)]
+struct InclusionProof {
+    // public input
```

Comment:
The public inputs of InclusionProof are `leaf_hash` and `root_hash`

Reply:
Yes, I know but the `root_hash` is already in the `mst`. 
I would prefer to avoid adding it redundantly.

Reply:
Agree